controller/linodefirewall_controller.go

```diff
@@ -115,7 +115,7 @@ func (r *LinodeFirewallReconciler) reconcile(
 		if err != nil {
 			fwScope.LinodeFirewall.Status.FailureReason = util.Pointer(failureReason)
 			fwScope.LinodeFirewall.Status.FailureMessage = util.Pointer(err.Error())
-			conditions.MarkFalse(fwScope.LinodeFirewall, clusterv1.ReadyCondition, string(failureReason), clusterv1.ConditionSeverityError, err.Error())
+			conditions.MarkFalse(fwScope.LinodeFirewall, clusterv1.ReadyCondition, string(failureReason), clusterv1.ConditionSeverityError, "%s", err.Error())
```

Comment:
Why are we adding `%s`?

Reply:
conditions.MarkFalse expects internally calls Message:  fmt.Sprintf(messageFormat, messageArgs...). err.Error() is working as string format instead of an actual string. If it has any unexpected % sign, this will fail. So golangci-lint has got picker with such things. %s works as a format string.